core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt

```diff
@@ -271,36 +282,48 @@ private fun makeChunk(
 ) {
     if (startOffset == endOffset)
         return
-    val rangeViewForward = TrackRangeView(startOffset.meters, endOffset.meters,
-        DiTrackEdgeImpl(track, Direction.FORWARD))
-    val rangeViewBackward = TrackRangeView(startOffset.meters, endOffset.meters,
-        DiTrackEdgeImpl(track, Direction.BACKWARD))
-    fun <T>makeDirectionalMap(f: (range: TrackRangeView) -> T) : DirectionalMap<T>{
+    val rangeViewForward = TrackRangeView(
+        startOffset.meters, endOffset.meters,
+        DiTrackEdgeImpl(track, Direction.FORWARD)
+    )
+    val rangeViewBackward = TrackRangeView(
+        startOffset.meters, endOffset.meters,
+        DiTrackEdgeImpl(track, Direction.BACKWARD)
+    )
+
+    fun <T> makeDirectionalMap(f: (range: TrackRangeView) -> T): DirectionalMap<T> {
         return DirectionalMap(f.invoke(rangeViewForward), f.invoke(rangeViewBackward))
     }
+
     fun makeSpeedSections(range: TrackRangeView): DistanceRangeMap<SpeedSection> {
         val res = distanceRangeMapOf<SpeedSection>()
         for (entry in range.speedSections.asMapOfRanges()) {
             val legacySpeedLimit = entry.value
             val map = legacySpeedLimit.speedLimitByTag
-                .mapValues { mapEntry -> Speed.fromMetersPerSeconds(mapEntry.value!!) }
+                .mapValues { mapEntry -> Speed.fromMetersPerSecond(mapEntry.value!!) }
             res.put(
                 Distance.fromMeters(entry.key.lowerEndpoint()),
                 Distance.fromMeters(entry.key.upperEndpoint()),
-                SpeedSection(Speed.fromMetersPerSeconds(legacySpeedLimit.defaultSpeedLimit), map)
+                SpeedSection(Speed.fromMetersPerSecond(legacySpeedLimit.defaultSpeedLimit), map)
             )
         }
         return res
     }
+
     fun makeNeutralSection(range: TrackRangeView): DistanceRangeMap<SimNeutralSection> {
         val res = distanceRangeMapOf<SimNeutralSection>()
-        for (entry in range.neutralSections.asMapOfRanges()) {
-            val legacyNeutralSection = entry.value
-            res.put(
-                Distance.fromMeters(entry.key.lowerEndpoint()),
-                Distance.fromMeters(entry.key.upperEndpoint()),
-                SimNeutralSection(legacyNeutralSection.lowerPantograph)
-            )
+        for ((rangeMap, isAnnouncement) in listOf(
+            Pair(range.neutralSections, false),
+            Pair(range.neutralSectionAnnouncements, true),
+        )) {
+            for (entry in rangeMap.asMapOfRanges()) {
+                val legacyNeutralSection = entry.value
+                res.put(
+                    Distance.fromMeters(entry.key.lowerEndpoint()),
+                    Distance.fromMeters(entry.key.upperEndpoint()),
+                    SimNeutralSection(legacyNeutralSection.lowerPantograph, isAnnouncement)
+                )
+            }
```

Comment:
this stuff seems unrelated

Reply:
We're now taking into account both the actual neutral sections as well as the announced neutral sections. The change makes sense to me. What do you mean exactly ?